src/ol/tilegrid/tilegrid.js

```diff
@@ -541,9 +541,7 @@ ol.tilegrid.createForExtent = function(extent, opt_maxZoom, opt_tileSize, opt_co
  * @api
  */
 ol.tilegrid.createXYZ = function(opt_options) {
-  var options = /** @type {olx.tilegrid.TileGridOptions} */ ({});
-  ol.object.assign(options, opt_options !== undefined ?
-      opt_options : /** @type {olx.tilegrid.XYZOptions} */ ({}));
+  var options = ol.object.assign({}, opt_options ? opt_options : {});
```

Comment:
`options` can't be a `olx.tilegrid.TileGridOptions` because it has the `maxZoom` property